compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -420,7 +420,7 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
         Kon(pat.tpe.stripAnnots, pats.map(pat => project(pat)))
       else if (fun.symbol.owner == scalaSeqFactoryClass && fun.symbol.name == nme.unapplySeq)
         projectList(pats)
-      else if (fun.symbol.info.resultType.isRef(scalaSomeClass))
+      else if (!fun.symbol.info.finalResultType.isRef(scalaOptionClass))
```

Comment:
I don't think this is correct in general, the return type of an option-less patmat may define a `def isEmpty: Boolean` method, if this method return `false`, then the case will not be taken.